java-frontend/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -2020,24 +1996,8 @@ private static ArrayTypeTreeImpl newArrayTypeTree(Optional<List<Tuple<InternalSy
   }
 
   private static void completeTypeTreeWithAnnotations(TypeTree type, Optional<List<AnnotationTreeImpl>> annotations) {
-    if (annotations.isPresent()) {
-      List<AnnotationTree> typeAnnotations = ImmutableList.copyOf(annotations.get());
-      completeTypeTreeWithAnnotations(type, typeAnnotations);
-    }
-  }
-
-  private static void completeTypeTreeWithAnnotations(TypeTree type, List<AnnotationTree> typeAnnotations) {
-    if (type.is(Tree.Kind.IDENTIFIER)) {
-      ((IdentifierTreeImpl) type).complete(typeAnnotations);
-    } else if (type.is(Tree.Kind.MEMBER_SELECT)) {
-      ((MemberSelectExpressionTreeImpl) type).complete(typeAnnotations);
-    } else if (type.is(Tree.Kind.PARAMETERIZED_TYPE)) {
-      ((ParameterizedTypeTreeImpl) type).complete(typeAnnotations);
-    } else if (type.is(Kind.ARRAY_TYPE)) {
-      ((ArrayTypeTreeImpl) type).complete(typeAnnotations);
-    } else {
-      ((PrimitiveTypeTreeImpl) type).complete(typeAnnotations);
-    }
+    List<AnnotationTree> annotationList = ImmutableList.copyOf(annotations.or(ImmutableList.of()));
```

Comment:
Why a copy of the list ?

Reply:
Because on all the calls to that method, the list was copied before. I simplify moved the copy from the caller to the callee.